srcts/src/components/sidebar.ts

```diff
@@ -297,24 +326,61 @@ class Sidebar {
     });
   }
 
+  /**
+   * Retrieves the current window size by reading a CSS variable whose value is
+   * toggled via media queries.
+   * @returns The window size as `"desktop"` or `"mobile"`, or `""` if not
+   * available.
+   */
+  private _getWindowSize(): SidebarWindowSize | "" {
+    const { container } = this.layout;
+
+    return window
+      .getComputedStyle(container)
+      .getPropertyValue("--bslib-sidebar-js-window-size")
+      .trim() as SidebarWindowSize | "";
+  }
+
   /**
    * Initialize the sidebar's initial state when `open = "desktop"`.
    * @private
    */
-  private _initDesktop(): void {
+  private _initSidebarState(): void {
     const { container } = this.layout;
-    // If sidebar is marked open='desktop'...
-    if (container.dataset.bslibSidebarOpen?.trim() !== "desktop") {
+
+    const initDesktop = container.dataset.openDesktop?.trim();
+    const initMobile = container.dataset.openMobile?.trim();
```

Comment:
Maybe put in a getter and do the `as` thing for typing

Reply:
I reorganized into helper methods and it's a lot more straightforward now, thanks!